### PR TITLE
Clone from jenkins-jobs repository for job updates

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Related we also have [duffy.groovy](https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/lib/duffy.groovy) and [foreman_infra.groovy](https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/lib/foreman_infra.groovy). For this job you should also update this line (and maybe renamed the whole job/file):
https://github.com/theforeman/jenkins-jobs/blob/fa60d415a7b30d981a79c800caf0b64ac9e89c13/theforeman.org/yaml/jobs/foreman-infra-jenkins-jobs-update.yml#L7

**Comment by ehelms** [comment]
Got all the other locations covered now

File: theforeman.org/yaml/jobs/foreman-infra-jenkins-jobs-update.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     sandbox: true
     properties:
       - github:
-          url: https://github.com/theforeman/foreman-infra
+          url: https://github.com/theforeman/jenkins-jobs
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/foreman-infra-jenkins-jobs-update.yml:7]: It feels odd to keep this job named `foreman-infra` while the job has been changed to a different repo.

File: theforeman.org/pipelines/lib/foreman_infra.groovy [hunk 2]
```diff
@@ -2,7 +2,7 @@ void git_clone_foreman_infra(args = [:]) {
     target_dir = args.target_dir ?: ''
```
Comment (ekohl) [inline theforeman.org/pipelines/lib/foreman_infra.groovy:2]: It's odd to name it `git clone foreman-infra` and then you end up with `jenkins-jobs`

File: theforeman.org/pipelines/lib/duffy.groovy [hunk 3]
```diff
@@ -1,8 +1,8 @@
 def provision() {
     fix_ansible_config()
-    git_clone_foreman_infra(target_dir: 'foreman-infra')
+    git_clone_foreman_infra(target_dir: 'jenkins-jobs')
 
-  	dir('foreman-infra/ci/centos.org/ansible') {
+  	dir('centos.org/ansible') {
```
Comment (ekohl) [inline theforeman.org/pipelines/lib/duffy.groovy:5]: I think this needs to continue using the target dir in the path

File: theforeman.org/pipelines/lib/duffy.groovy [hunk 4]
```diff
@@ -15,19 +15,19 @@ def fix_ansible_config() {
 }
 
 def deprovision() {
-    if (fileExists('foreman-infra')) {
-        dir('foreman-infra/ci/centos.org/ansible') {
+    if (fileExists('jenkins-jobs')) {
+        dir('centos.org/ansible') {
```
Comment (ekohl) [inline theforeman.org/pipelines/lib/duffy.groovy:19]: Same here

File: centos.org/pipelines/lib/foremanCentosJob.groovy [hunk 5]
```diff
@@ -62,7 +62,7 @@ pipeline {
                 extra_vars = buildExtraVars(extraVars: playBook['extraVars'])
                 duffy_ssh("cd forklift && ansible-playbook playbooks/collect_debug.yml --limit '${playBook['boxes'].join(',')}' ${extra_vars}", 'duffy_box', './')
                 runPlaybook(
-                    playbook: 'foreman-infra/ci/centos.org/ansible/fetch_debug_files.yml',
+                    playbook: 'centos.org/ansible/fetch_debug_files.yml',
```
Comment (ekohl) [inline centos.org/pipelines/lib/foremanCentosJob.groovy:65]: I think this also needs to contain the directory name

File: theforeman.org/yaml/jobs/foreman-infra-jenkins-jobs-update.yml (hunk 1 above)
Comment (ehelms) [reply theforeman.org/yaml/jobs/foreman-infra-jenkins-jobs-update.yml:7]: The intent was not to keep it that way long term, I have been trying to do this in stages this whole time to reduce the amount of places we might break something.

File: theforeman.org/pipelines/lib/foreman_infra.groovy [hunk 6]
```diff
@@ -2,7 +2,15 @@ void git_clone_foreman_infra(args = [:]) {
...
 
     dir(target_dir) {
-        git url: 'https://github.com/theforeman/foreman-infra', poll: false
+        git url: 'https://github.com/theforeman/foreman_infra', poll: false
+    }
+}
+
+void git_clone_jenkins_jobs(args = [:]) {
```
Comment (ehelms) [inline theforeman.org/pipelines/lib/foreman_infra.groovy:9]: Added a new method for this repository given that some jobs need the puppet part of foreman-infra.

**Review by ekohl — APPROVED** [review]

**Pull request merged**